src/rsz/src/Resizer.cc

```diff
@@ -1165,7 +1168,15 @@ Resizer::dbuToMeters(int dist) const
 int
 Resizer::metersToDbu(double dist) const
 {
-  return dist * dbu_ * 1e+6;
+  if (dist < 0) {
+    logger_->error(
+        RSZ, 86, "metersToDbu({}) cannot convert negative distances", dist);
+  }
+  // sta::INF is passed to this function in some cases. Protect against
+  // overflow conditions.
+  double distance = dist * dbu_ * 1e+6;
+  return static_cast<int>(std::lround(distance)
+                          & std::numeric_limits<int>::max());
```

Comment:
Nothing about unit conversion prevents a negative value from being converted.  Only underflow / overflow is an issue.  I think std::clamp 'ing to the integer min()/max() would be more correct and avoid the bit trickery.

Reply:
This function accepts a distance which conceptually shouldn't be negative.

I would rather have callers call abs on their code if it is. I'll change to std::clamp

Reply:
The argument name suggests distance but the actual operation is just metersToDBU which can operate on coordinates as well as distances.  I think the argument name should be changed.  If the incoming values shouldn't be negative that's on the caller.